source/common/stats/statsd.cc

```diff
@@ -33,18 +33,19 @@ Writer::~Writer() {
   }
 }
 
-void Writer::writeCounter(const std::string& name, uint64_t increment) {
-  std::string message(fmt::format("envoy.{}:{}|c", name, increment));
+void Writer::writeCounter(const std::string& name, const std::string& tag_str, uint64_t increment) {
+  std::string message(fmt::format("envoy.{}:{}|c{}\n", name, increment, tag_str));
```

Comment:
Should we avoid the `\n` for regular `statsd`? i.e. only add it in `buildTagStr()` if `use_tag_`?

Reply:
Nice point. Actually we don't need trailing newline, so just dropped it. b664048e1dfc3b9c4e6444a1097a5f69dd892c79